pkg/api/v1/types.go

```diff
@@ -1709,6 +1714,15 @@ type SecretType string
 
 const (
 	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+
+	// SecretTypeDockercfg contains a dockercfg file that follows the same format rules as ~/.dockercfg
+	//
+	// Required fields:
+	// - Secret.Data["dockercfg"] - a serialized ~/.dockercfg file
+	SecretTypeDockercfg SecretType = "kubernetes.io/dockercfg"
+
+	// DockerConfigKey is the key of the required data for SecretTypeDockercfg secrets
+	DockerConfigKey = ".dockercfg"
```

Comment:
> Do we really need the leading dot?

The leading dot matches the current, default filename exactly.  It makes it very convenient to directly bundle a secret from an existing file and means less transformation if a pod wants to mount it and use it as is.